src/turnin.c

```diff
@@ -1364,14 +1380,26 @@ void checkdue() {
 	diff_days = (int)diff_time / 86400; // in days
 
 	// calculate penalty
-	tm_curr = localtime(&curr_time);
-	penalty += (tm_curr->tm_wday == 0 || tm_curr->tm_wday == 6) ? weekendpenalty : daypenalty;
+	res = localtime_r(&curr_time, &tm_curr);
+	if (!res) {
+		fprintf(stderr,	
+				"turnin: Cannot create timestamp\n"
```

Comment:
Maybe make this line different between the 3 cases that can result in it, so you can figure out where exactly was the problem in the code.